test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala

```diff
@@ -51,43 +51,70 @@ class QueueControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplic
     }
 
     val mockQueueService = mock[QueueService]
-    val queueController = new QueueController(mockQueueService, new StaticIDGenerator)
+    val mockFieldsService = mock[ApiSubscriptionFieldsService]
+    val queueController = new QueueController(mockQueueService, mockFieldsService, new StaticIDGenerator)
   }
 
   "POST /queue" should {
-    "throw exception when then X-Client-ID header is not sent in the request" in new Setup {
-      intercept[BadRequestException] {
-        await(queueController.save()(FakeRequest(POST, "/queue")))
-      }
+    "return 400 when then X-Client-ID header is not sent in the request" in new Setup {
```

Comment:
I believe we need to update the title of this test now.
I'd say something like this:
"return 400 when any of the `X-Client-ID` and `api-subscription-fields-id` headers are sent in the request"
Do you agree?